lib/media/track/remotevideotrack.js

```diff
@@ -3,8 +3,10 @@
 const mixinRemoteMediaTrack = require('./remotemediatrack');
 const VideoTrack = require('./videotrack');
 const documentVisibilityMonitor = require('../../util/documentvisibilitymonitor.js');
+const NullIntersectionObserver = require('../../util/nullintersectionobserver.js');
 
 const RemoteMediaVideoTrack = mixinRemoteMediaTrack(VideoTrack);
+const RealIntersectionObserver =  typeof IntersectionObserver !== 'undefined' ? IntersectionObserver : NullIntersectionObserver;
```

Comment:
I suggest lets remove `RealIntersectionObserver` variable and assign the observer directly like in the constructor

```
    options = Object.assign({
      idleTrackSwitchOff: true,
      enableDocumentVisibilityTurnOff: true,
      IntersectionObserver: typeof IntersectionObserver !== 'undefined' ? IntersectionObserver : NullIntersectionObserver,
    }, options);
```